plugins/metalink/utils.c

```diff
@@ -631,83 +644,70 @@ TDNFParseUrlTag(
                                  (void**)&ml_url_info);
     BAIL_ON_TDNF_ERROR(dwError);
 
-    if ((xmlPropValue = xmlGetProp(node, ATTR_PROTOCOL)))
-    {
-        value = (const char*)xmlPropValue;
-        dwError = TDNFAllocateString(value, &(ml_url_info->protocol));
-        BAIL_ON_TDNF_ERROR(dwError);
-        xmlFree(xmlPropValue);
-        xmlPropValue = NULL;
-    }
-    if ((xmlPropValue = xmlGetProp(node, ATTR_TYPE)))
-    {
-        value = (const char*)xmlPropValue;
-        dwError = TDNFAllocateString(value, &(ml_url_info->type));
-        BAIL_ON_TDNF_ERROR(dwError);
-        xmlFree(xmlPropValue);
-        xmlPropValue = NULL;
-    }
-    if ((xmlPropValue = xmlGetProp(node, ATTR_LOCATION)))
-    {
-        value = (const char*)xmlPropValue;
-        dwError = TDNFAllocateString(value, &(ml_url_info->location));
-        BAIL_ON_TDNF_ERROR(dwError);
-        xmlFree(xmlPropValue);
-        xmlPropValue = NULL;
-    }
-    if ((xmlPropValue = xmlGetProp(node, ATTR_PREFERENCE)))
+    for (int i = 0; elementInfo->attributes[i]; i += 2)
     {
-        value = (const char*)xmlPropValue;
-        if(sscanf(value, "%d", &prefValue) != 1)
+        if (!strcmp(elementInfo->attributes[i], ATTR_PROTOCOL))
         {
-            dwError = ERROR_TDNF_INVALID_PARAMETER;
-            pr_err("XML Parser Warning: Preference is invalid value: %s\n", value);
-            BAIL_ON_TDNF_ERROR(dwError);
+           value = (char *)elementInfo->attributes[i + 1];
```

Comment:
isnt this going to crash if attributes key is present but value is missing, Its better to check atributes[i+1] is not NULL.If taking care of this please do it for wherever using this atributes[i+1]  directly.

Reply:
Done